src/main/java/seedu/address/model/game/DecreasingMode.java

```diff
@@ -0,0 +1,90 @@
+package seedu.address.model.game;
+
+import java.util.Date;
+
+import seedu.address.model.task.Task;
+
+// @@author chikchengyao
+
+/**
+ * Offers a mode where full XP is awarded for tasks done more than a certain time period
+ * before the deadline, measured in days.
+ */
+public class DecreasingMode extends GameMode {
+
+    private int daysBefore;
+    private int overdueXp;
+    private int completedXp;
+
+    DecreasingMode() {
+        this(7, 25, 50);
+    }
+
+    DecreasingMode(int daysBefore, int overdueXp, int completedXp) {
+        this.daysBefore = daysBefore;
+        this.overdueXp = overdueXp;
+        this.completedXp = completedXp;
+    }
+
+    @Override
+    public int appraiseXpChange(Task taskFrom, Task taskTo) {
```

Comment:
Consider adding some header comments to explain what does overdueXP and completedXP mean, and in which scenarios are they awarded to make this method easier to understand.